packages/mongodb-chatbot-server/src/routes/responses/createResponse.test.ts

```
@@ -387,154 +432,186 @@ describe("POST /responses", () => {
-      expect(addedMessages[1].role).toEqual("system");
+      expect(updatedConversation.messages[0].role).toEqual("system");
+      expect(updatedConversation.messages[0].content).toEqual(functionCallType);
 
-      expect(addedMessages[0].content).toEqual(functionCallType);
-      expect(addedMessages[1].content).toEqual(functionCallOutputType);
+      expect(updatedConversation.messages[1].role).toEqual("system");
+      expect(updatedConversation.messages[1].content).toEqual(
+        functionCallOutputType
+      );
     });
   });
 
   describe("Invalid requests", () => {
     it("Should return 400 with an empty input string", async () => {
-      const response = await makeCreateResponseRequest({
+      const { response } = await makeClientAndRequest({
         input: "",
       });
+      const results = await collectStreamingResponse(response);
 
-      expect(response.statusCode).toBe(400);
-      expect(response.body.error).toEqual(
-        badRequestError(`Path: body.input - ${ERR_MSG.INPUT_STRING}`)
-      );
+      expect(response.status).toBe(200);
+      testInvalidResponses({
+        responses: results,
+        message: `Path: body.input - ${ERR_MSG.INPUT_STRING}`,
+      });
     });
 
     it("Should return 400 with an empty message array", async () => {
-      const response = await makeCreateResponseRequest({
+      const { response } = await makeClientAndRequest({
         input: [],
       });
+      const results = await collectStreamingResponse(response);
 
-      expect(response.statusCode).toBe(400);
-      expect(response.body.error).toEqual(
-        badRequestError(`Path: body.input - ${ERR_MSG.INPUT_ARRAY}`)
-      );
+      expect(response.status).toBe(200);
+      testInvalidResponses({
+        responses: results,
+        message: `Path: body.input - ${ERR_MSG.INPUT_ARRAY}`,
+      });
     });
 
     it("Should return 400 if model is not mongodb-chat-latest", async () => {
```

nit on name: rather than hardcoding this in, pls make it more generic, like "Should return error if model is not supported in configuration" (or smthn like that, leave naming up to you)